Sources/SKSwiftPMWorkspace/SwiftPMWorkspace.swift

```diff
@@ -416,12 +400,22 @@ extension SwiftPMWorkspace {
   }
 
   /// Retrieve settings for a given header file.
+  ///
+  /// This finds the target the header belongs to based on its location in the file system and returns
+  /// the build settings for any file within that target. This is safe because all files within one target
+  /// have the same build settings.
```

Comment:
Is there some reason the API takes file at all then?

Reply:
CC @neonichu

Reply:
The arguments for C source files contain the source file name as well, so technically they're not 100% the same :)

Reply:
Ah, I changed the call to patch the source file, ie. replacing the arbitrarily picked source file with the header file and a `-x` flag to specify the header file’s language.

@neonichu Could you take a quick look at the new implementation and see if it (and the comments) make sense to you?

Reply:
> The arguments for C source files contain the source file name as well, so technically they're not 100% the same :)

Reasonable point, though we could always it 🤷‍♂️ 

> Ah, I changed the call to patch the source file, ie. replacing the arbitrarily picked source file with the header file and a -x flag to specify the header file’s language.

Does SPM allow mixed language targets? Also, regarding the API @neonichu is there any guarantee on the order of `sources`?

Reply:
I believe sources are sorted lexicographically upstream, but I haven't defined any explicit guarantees on the order for this API.